### ISSUE TITLE
RegExp slash escapes not parsed correctly

### ISSUE THREAD

**Issue by cartant** [estools/esquery#68]
Using the following snippet in the [ESQuery demo](http://estools.github.io/esquery/):

```js
var f = "foo/bar";
```

None of these queries is parsed correctly and all fail with an error:

`[value=/foo\/bar/]`

> Invalid regular expression: /foo\/: \ at end of pattern

`[value=/foo\\/bar/]`

> Expected " " or "]" but "b" found.

`[value=/foo\\\\/bar/]`

> Expected " " or "]" but "b" found.

I'm using [`tsquery`](https://github.com/phenomnomnominal/tsquery) to write an import-location-specific TSLint rule for RxJS and the RegExp issue appears to be an upstream problem with this package.

**Comment by michaelficarra** [comment]
Yeah, the regexp parsing leaves a bit to be desired: https://github.com/estools/esquery/blob/1853a6fbaf40709dc2bf9f3c027163a573c3824f/grammar.pegjs#L83

We should enhance it to at least take into account escape sequences and character classes.

**Comment by cartant** [comment]
My situation was easy to work around, so this is not something that's blocking me. I can have a look at it later, if you are interested in a PR to fix the problem.

**Comment by michaelficarra** [comment]
Sure, send one over.

**Comment by toefraz** [comment]
As a workaround you can use the unicode code for `/` in place of the actual character:

`[value=/foo\\u002Fbar/]`

**Comment by colinta** [comment]
@toefraz I "optimized" your code. 🤣 `[value=/foo\\x2Fbar/]` (I _am_ kidding, but also I _am_ flexing my knowledge of obscure JS string parsing)

**Comment by pke** [comment]
Will this ever be fixed?
